src/js/tabs/firmware_flasher.js

```diff
@@ -817,20 +813,39 @@ firmware_flasher.initialize = function (callback) {
                     if ($('input.flash_manual_baud').is(':checked')) {
                         baud = parseInt($('#flash_manual_baud_rate').val());
                     }
-                    console.log('Query board information to preselect right firmware');
+                    GUI.log(i18n.getMessage('firmwareFlasherDetectBoardQuery'));
                     if (!(serial.connected || serial.connectionId)) {
                         serial.connect(port, {bitrate: baud}, onConnect);
                     } else {
                         console.warn('Attempting to connect while there still is a connection', serial.connected, serial.connectionId);
                         serial.disconnect();
                     }
                 } else {
-                    console.log('Please select valid serial port');
                     GUI.log(i18n.getMessage('firmwareFlasherNoValidPort'));
                 }
             }
         }
 
+        const detectBoardElement = $('a.detect-board');
+
+        detectBoardElement.on('click', verifyBoard);
+
+        function updateDetectBoardButton() {
+            const board = $('select[name="board"] option:selected').val();
+            const firmwareVersion = $('select[name="firmware_version"] option:selected').val();
+            const isDfu = portPickerElement.val().includes('DFU');
```

Comment:
wondering if "DFU" stays as "DFU" in all other languages for the element value?

Reply:
@limonspb DFU is not translated.

Reply:
'DFU' is a term STM uses in their documentation, so I doubt this should be translated.

Reply:
I don't know if some alphabets, like Cyrillic, Chinese, Japanese, etc. can have some different symbols for the same.